### PR TITLE
feat(template):detach on complete and error

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR tries to evaluate the pros/cons edgecases of detaching the EmbeddedView after error or complete.

### CODE REVIEW

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
this is a very nice idea. but I think we need to move this into a strategy or something. it will lead to problems, when changes from outside should get reflected in those templates.

just a little example:

```html

            <!-- does not work when template gets detached -->
            <ng-container>{{ errorTmplState$ | push }}</ng-container>
            <!-- works fine when template gets detached -->
            <ng-container *rxLet="errorTmplState$; let state">{{ state }}</ng-container>
```

**Comment by BioPhoton** [comment]
Moved it into the local strategies. I had to extend the work function with the current notification to know the active template

File: apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/default-strategies.ts [hunk 1]
```diff
@@ -18,7 +14,14 @@ export function getDefaultStrategyCredentialsMap(): StrategyCredentialsMap {
...
   name: 'local',
-  work: (cdRef) => cdRef.detectChanges(),
+  work: (cdRef, _, notification) => {
+    if(['rxComplete', 'rxError'].includes(notification.kind)) {
+      cdRef.detach();
+    } else {
+      cdRef.reattach();
+      cdRef.detectChanges();
+    }
+  },
```
Comment (hoebbelsB) [inline apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/default-strategies.ts:24]: so the error and complete template will not receive any changes? don't they receive some context which should get updated?
Also, I think this implementation is very slow, we should store the methods in a map and just call

```ts
notificationWorkMap[notification.kind]();
```

this way we save 1.000s of iterations over the same array again and again.

File: apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/model.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { Observable } from 'rxjs';
 import { ChangeDetectorRef } from '@angular/core';
+import { RxNotification } from '@rx-angular/template';
 
-export type RenderWork = <T = unknown>(cdRef: ChangeDetectorRef, context: any) => void;
+export type RenderWork = <T = unknown>(cdRef: ChangeDetectorRef, context?: any, notification?: RxNotification<T>) => void;
```
Comment (hoebbelsB) [inline apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/model.ts:5]: I think this can be quite helpful :)

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
do we really want to have the complete and error templates cold?

File: apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/default-strategies.ts (hunk 1 above)
Comment (BioPhoton) [reply apps/demos/src/app/shared/rx-angular-pocs/render-stragegies/default-strategies.ts:24]: "cold" was confusing at the beginning for me. But got it. 
You mean values from the observable. 

There is no next value after error or complete that could occur from the observable. 
And the triggers for the template has to behave the same. Detach temple and no other CDs affect it. A potential next value will reattach the view.

For the map:
Sure we have to make to nice! Fir the Poc it was important to see what we need to change in the strategy config.

So I will skip any changes on low level implementations.
